FreeRTOS_IP.c

```diff
@@ -3824,7 +3803,7 @@ void vReturnEthernetFrame( NetworkBufferDescriptor_t * pxNetworkBuffer,
         else if( ( uxMinLastSize * ipMONITOR_PERCENTAGE_90 ) > ( uxMinSize * ipMONITOR_PERCENTAGE_100 ) )
         {
             uxMinLastSize = uxMinSize;
-            FreeRTOS_printf( ( "Heap: current %d lowest %u\n", ( printf_signed ) xPortGetFreeHeapSize(), ( printf_unsigned ) uxMinSize ) );
+            FreeRTOS_printf( ( "Heap: current %d lowest %u\n", xPortGetFreeHeapSize(), uxMinSize ) );
```

Comment:
Last week I spent a lot of time to sort out how printf compiler warnings can be avoided while getting the format correct, see my comment in [PR #197](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/197#pullrequestreview-595091959). That works for both 32 and 64-bit platforms.

This is one example:
~~~
    uint32_t ulValue = 0x80004000U;
    printf( "int32 = %x\n", ( unsigned int ) ulValue );
~~~
Earlier, I have been playing with `printf_signed ` and some other typedef's, but Coverity didn't like them.
I would like to propose to use casts `( unsigned int )` as in the example, because that is what `%x` is expecting.

Reply:
That seems fine for unsigned values.

Reply:
In a meanwhile I also tested the formats with signed numbers. The outputs are as expected.